src/core/net/udp6.cpp

```diff
@@ -205,6 +221,35 @@ otError Udp::Bind(SocketHandle &aSocket, const SockAddr &aSockAddr)
     return error;
 }
 
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+void Udp::BindToNetif(SocketHandle &aSocket, otNetifIdentifier aNetifIdentifier)
+{
+    if (aNetifIdentifier == OT_NETIF_BACKBONE)
+    {
+        SetBackboneSocket(aSocket);
+    }
+}
+
+void Udp::SetBackboneSocket(SocketHandle &aSocket)
+{
+    RemoveSocket(aSocket);
+
+    if (mPrevBackboneSockets != nullptr)
+    {
+        mSockets.PushAfter(aSocket, *mPrevBackboneSockets);
+    }
+    else
+    {
+        IgnoreError(mSockets.Add(aSocket));
```

Comment:
Use  `mSocket.Push(aSocket)` here?

`Add()` checks for it not to be on the list already before adding/pushing it to the list but since we know it's not in the list (we did `RemoveSocke()` already) we can use `Push` (which does not check).

Reply:
revised